code/obj/item.dm

```diff
@@ -30,6 +30,10 @@
 	var/burn_type = 0 //0 = ash, 1 = melt
 	var/burning_last_process = 0
 	var/firesource = FALSE //TRUE or FALSE : dictates whether or not the item can be used as a valid source of fire
+	/// How many items on a given turf were hotboxed?
+	var/hotbox_count = null
+	/// Item used to hotbox
+	var/atom/hotbox_item = null
```

Comment:
hotbox_count is unneeded with the other suggested changes.

hotbox_item is hold an object ref which could prevent clean garbage collection. Instead I'd suggest you just store the fingerprints of that item directly since that's all you're using it for anyway, and it avoid object references. The name would probably need to be changed too to something that fits better with this change.